src/main/java/nl/rutgerkok/pokkit/world/PokkitWorld.java

```diff
@@ -358,8 +358,15 @@ public boolean getKeepSpawnInMemory() {
 
 	@Override
 	public List<LivingEntity> getLivingEntities() {
-		throw Pokkit.unsupported();
-
+		List<LivingEntity> livingEntities = new ArrayList<LivingEntity>();
+		
+		for (Entity entity : getEntities()) {
```

Comment:
I think this method uses more resources than necessary. This method calls getEntities, which allocates a List and Entity wrappers for all entities. Then this method creates another list, and throws many of the previously allocated entities out.

It think something like this would be better:

```javascript
List<LivingEntity> livingEntities = new ArrayList<>();
for (cn.nukkit.entity.Entity entity : nukkit.getEntities()) {
	if (Nukkit reports entity is a living entity) {
 		livingEntities.add((LivingEntity) PokkitEntity.toBukkit(entity));
	}
}
return livingEntities;
```